### PR TITLE
Backport NXP-28763 NXP-26704 NXP-28900 NXP-28919 to 10.10

### PR DESCRIPTION
**Pull request by efge**
The code is 99% identical to the one in branch `master`, except for Java 11 details like `Set.of()` and `String.repeat()`.

The goal of the review is mainly to identify backward-compatibility issues.

### CODE REVIEW

**Comment by efge** [comment]
T&P ok: https://qa2.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-fguillaume-10.10/86/

File: nuxeo-core/nuxeo-core-storage-mongodb/src/main/java/org/nuxeo/ecm/core/storage/mongodb/MongoDBConverter.java [hunk 1]
```diff
@@ -57,22 +64,35 @@
...
      * Constructor for a converter that also knows to optionally translate the native MongoDB "_id" into a custom id.
+     * <p>
+     * When {@code idValuesKeys} are provided, the ids are stored as longs.
      *
      * @param idKey the key to use to map the native "_id" in memory, if not {@code null}
+     * @param trueOrNullBooleanKeys the keys corresponding to boolean values that are only true or null (instead of
+     *            false)
+     * @param idValuesKeys the keys corresponding to values that are ids
      */
-    public MongoDBConverter(String idKey) {
```
Comment (troger) [inline nuxeo-core/nuxeo-core-storage-mongodb/src/main/java/org/nuxeo/ecm/core/storage/mongodb/MongoDBConverter.java:74]: Don't we want to keep the existing constructor/methods for backward compatibility, to be safe? (I know it's mostly internal stuff)

File: nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/blob/DocumentBlobManager.java [hunk 2]
```diff
@@ -35,14 +35,27 @@
...
+    /**
+     * Reads a {@link Blob} from storage.
+     *
+     * @param blobInfo the blob information
+     * @param doc the document to which this blob belongs
+     * @param xpath the xpath of the blob in the document
+     * @return a managed blob
+     * @since 11.1
+     */
+    Blob readBlob(BlobInfo blobInfo, Document doc, String xpath) throws IOException;
```
Comment (troger) [inline nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/blob/DocumentBlobManager.java:47]: Maybe a default impl?

File: nuxeo-core/nuxeo-core-storage-mongodb/src/main/java/org/nuxeo/ecm/core/storage/mongodb/MongoDBConverter.java (hunk 1 above)
Comment (efge) [reply nuxeo-core/nuxeo-core-storage-mongodb/src/main/java/org/nuxeo/ecm/core/storage/mongodb/MongoDBConverter.java:74]: This is too low-level internals to make sense.

File: nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/blob/DocumentBlobManager.java (hunk 2 above)
Comment (efge) [reply nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/blob/DocumentBlobManager.java:47]: As it's a service with only one implementation not really worth it, and it has to be implemented anyway.

**Pull request merged**